File: Programmers/Lv.0/리스트 자르기/181897.java
class Solution {
    public int[] solution(int n, int[] slicer, int[] num_list) {
        int[] answer = {};
        
        int a = slicer[0], b = slicer[1], c = slicer[2];
        int idx = 0;
        
        if (n == 1) {
            answer = new int[b + 1];
            for (int i=0; i<=b; i++) answer[i] = num_list[i];
            
        } else if (n == 2) {
            answer = new int[num_list.length - a];
            for (int i=a; i<num_list.length; i++) answer[idx++] = num_list[i];
            
        } else if (n == 3) {
            answer = new int[b - a + 1];
            for (int i=a; i<=b; i++) answer[idx++] = num_list[i];
            
        } else {
            answer = new int[(b - a) / c + 1];
            for (int i=a; i<=b; i+=c) answer[idx++] = num_list[i];
        }
        
        return answer;
    }
}

File: Programmers/Lv.0/리스트 자르기/181897.js
function solution(n, slicer, num_list) {
    const answer = [];
    
    const [a, b, c] = slicer;
    const start = (n === 1) ? 0 : a;
    const end = (n === 2) ? num_list.length-1 : b;
    const step = (n === 4) ? c : 1;
    
    for (let i=start; i<=end; i+=step) {
        answer.push(num_list[i]);
    }
    
    return answer;
}
